scripts/download_model_urls.py

```diff
@@ -0,0 +1,41 @@
+import asyncio
```

Comment:
This file is a implementation of 

https://github.com/pytorch/vision/blob/5850f370c03d941f97c7bd53f99a83abb0b9dd01/.circleci/config.yml#L171-L189

in Python. The old version relied on `wget` and `parallel` installed through `apt`, but they are not available through `conda`.

Reply:
One difference is that this PR uses async downloads, while the old version used multiprocessing. It seems async is roughly 5x slower:

- multiprocessing 1m 2s: https://app.circleci.com/pipelines/github/pytorch/vision/23876/workflows/d96da5f3-9ca0-4615-9c08-0373c00233a0/jobs/1849889
- async 5m 2s: https://github.com/pytorch/vision/actions/runs/4363141530/jobs/7628874383#step:10:776

I'll try multiprocessing and see if this actually is the root cause or this just comes from the environment change between CircleCI and GHA.

Reply:
I've tried multiprocessing with threads in 5d6f391. The run aborted to a `MemoryError`. From the logs we can see though that it also took over 5 minutes: https://github.com/pytorch/vision/actions/runs/4364016074/jobs/7630816354#step:10:894

Thus, I would go with the async solution since that worked. I'm no expert in async / multiprocessing though. If someone sees possible perf improvements for either implementations, feel free to suggest.

Reply:
I've tried the solution with `wget` and `parallel` on GHA and it seems it is really the env that is causing the slowdown:

```
$ time python scripts/collect_model_urls.py torchvision/models/ | parallel -j0 'wget --no-verbose -O foo/`basename {}` {}\?source=ci'
[...]
real    5m0.152s
user    0m49.044s
sys     1m10.467s
```

Meaning, I'm totally fine using the async solution.